builder/optimizations.cc

```diff
@@ -941,6 +959,12 @@ class in_place_aliaser : public stmt_mutator {
       set_result(block::make(std::move(stmts)));
     }
   }
+
+  void visit(const loop* op) override {
+    ++loop_level;
```

Comment:
Would it work for case like this:
```
allocate v {
for (x) {
   call_stmt(..., v);
}
for (x) {
   v_aliased = crop_buffer(v);
   call_stmt(..., v_aliased);
   call_stmt(v_aliased, ...);
} 
```
? Or is this something aliaser wouldn't do anyway?

Reply:
I think we don't want to do this kind of aliasing unless we can prove that the written values aren't used across loop iterations, which we could do if we tracked bounds in this visitor but we don't currently. The first loop is really irrelevant here, it's just that the allocation is outside the loop that makes things hard.

Reply:
Right, but wouldn't they have the same loop levels in this case, so this fix wouldn't prevent it from happening?

Reply:
The calls have the same loop level, but the allocation doesn't. This fix doesn't actually compare the loop levels of two different calls at all, only the loop level of the input and output allocations.

Reply:
ah, that's right, sorry.